src/Pate/Proof/CounterExample.hs

```diff
@@ -182,27 +204,30 @@ getPathCondition ::
   forall sym arch.
   (MM.RegisterInfo (MM.ArchReg arch)) =>
   PS.SimBundle sym arch ->
-  PF.BlockSliceState (PFI.ProofSym sym arch) ->
-  PFI.SymDomain sym arch ->
+  PF.BlockSliceState sym arch ->
+  PF.EquivalenceDomain sym arch ->
   SymGroundEvalFn sym ->
   EquivM sym arch (PPa.PatchPairC (W4.Pred sym))
 getPathCondition bundle slice dom fn = withSym $ \sym -> do
-  groundDom <- groundProofExpr fn dom
   let
+    regInDomain ::
+      MM.ArchReg arch tp -> EquivM sym arch Bool
+    regInDomain reg =
+      execGroundFn fn $ getConst $ PF.eqDomainRegisters dom ^. MM.boundValue reg
+
     getRegPath ::
```

Comment:
This might be something to add in a different PR (perhaps in the upcoming cleanups), but what is the intent in the logic here with `regInDomain`? What is the intent of `getRegPath` (especially in the case where `not (regInDomain)`?

Reply:
This just avoids examining the terms for registers which are outside of the equivalence domain (i.e. they were not part of the goal equivalence proof and therefore should not be examined when trying to come up with a path condition)